### ISSUE TITLE
Marker-level events not firing after restoring old state

### ISSUE THREAD

**Issue by KedroX** [ailon/markerjs2#183]
Hi,

I don't know if it's a bug or simply not implemented but I'm trying to use Marker-level events to handle marker change. It works as expected for new created markers during current editing sessions, but not after restoring an old state. In this case events are not firing. 

Is there a way to do this ? 

I'm using marker.js 2 from UNPKG CDN 

`https://unpkg.com/markerjs2/markerjs2.js`

Here is an example with "markerchange" event. After restoring state, if you modify a restored marker from old state, event is not fired : 

```
import "./styles.css";
import * as markerjs2 from "markerjs2";

let sourceImage, targetRoot, maState;

// save references to the original image and its parent div (positioning root)
function setSourceImage(source) {
  sourceImage = source;
  targetRoot = source.parentElement;
}

function showMarkerArea(target) {
  const markerArea = new markerjs2.MarkerArea(sourceImage);
  // since the container div is set to position: relative it is now our positioning root
  // end we have to let marker.js know that
  markerArea.targetRoot = targetRoot;
  markerArea.addEventListener("render", (event) => {
    target.src = event.dataUrl;
    // save the state of MarkerArea
    maState = event.state;
  });
  markerArea.show();
  // if previous state is present - restore it
  if (maState) {
    markerArea.restoreState(maState);
  }

  markerArea.addEventListener("markerchange", (event) => {
    alert("change");
  });
}

setSourceImage(document.getElementById("sourceImage"));
```

Thanks a lot for your help! And congratulations for this good library !

L.

**Comment by ailon** [comment]
Hmm... Not sure what could be wrong here. It's very similar to the [save and restore state demo](https://codesandbox.io/p/sandbox/marker-js-2-save-and-restore-user-marker-settings-hyntdx?file=%2Fsrc%2Findex.js) and it works there. Do you get any errors in the console maybe?

**Comment by KedroX** [comment]
Hi,
Yes it works in the demo, but only during the current editing session. I just added an alert in the markerchange event to check. Here are the steps to reproduce : 
- click image to show marker area
- create a rectangle 
- validate and close the marker area by clicking the V at the top right
- click again image to show marker area
- the modify the rectangle created before
- Event markerchange is not fired. But it's fired if I create a new rectangle, not for the for the first marker created before
Thanks for your help.

**Comment by ailon** [comment]
Oh, I get and see it now. Thank you for the details! I will investigate and fix this.

**Comment by ailon** [comment]
This should be fixed in 2.31.1. Thank you for reporting this bug, @KedroX!

**Comment by KedroX** [comment]
Thanks to you for your job!
